tests/test_interpolatedimage.py

```diff
@@ -1036,6 +1036,72 @@ def test_stepk_maxk():
     t2 = time.time()
     print 'time for %s = %.2f'%(funcname(),t2-t1)
 
+def test_kround_trip():
+    a = final
+    real_a, imag_a = a.drawKImage()
+    b = galsim.InterpolatedKImage(real_a, imag_a)
+
+    # Check picklability
+    do_pickle(b)
+    do_pickle(b, lambda x: x.drawImage())
+    # do_pickle(b.SBProfile)
+    # do_pickle(b.SBProfile, lambda x: repr(x))
+
+    for kx, ky in zip(KXVALS, KYVALS):
+        np.testing.assert_almost_equal(a.kValue(kx, ky), b.kValue(kx, ky), 2,
+            err_msg=("InterpolatedKImage evaluated incorrectly at ({:0},{:1})"
+                     .format(kx, ky)))
+
+    np.testing.assert_almost_equal(a.getFlux(), b.getFlux(), 6) #Fails at 7th decimal
+
+    real_b, imag_b = b.drawKImage(real_a.copy(), imag_a.copy())
+    # Fails at 4th decimal
+    np.testing.assert_array_almost_equal(real_a.array, real_b.array, 3,
+                                         "InterpolatedKImage kimage drawn incorrectly.")
+    # Fails at 4th decimal
+    np.testing.assert_array_almost_equal(imag_a.array, imag_b.array, 3,
+                                         "InterpolatedKImage kimage drawn incorrectly.")
+
+    img_a = a.drawImage()
+    img_b = b.drawImage(img_a.copy())
+    # This is the one that matters though; fails at 6th decimal
+    np.testing.assert_array_almost_equal(img_a.array, img_b.array, 5,
+                                         "InterpolatedKImage image drawn incorrectly.")
+
+    # Try some (slightly larger maxk) non-even kimages:
+    for dx, dy in zip((2,3,3), (3,2,3)):
+        shape = real_a.array.shape
+        real_a, imag_a = a.drawKImage(nx=shape[1]+dx, ny=shape[0]+dy, scale=real_a.scale)
+        b = galsim.InterpolatedKImage(real_a, imag_a)
+
+        np.testing.assert_almost_equal(a.getFlux(), b.getFlux(), 6) #Fails at 7th decimal
+        img_b = b.drawImage(img_a.copy())
+        # One of these fails in 0.02% of pixels at 6th decimal
+        np.testing.assert_array_almost_equal(img_a.array, img_b.array, 5)
+
+    # Try some additional transformations:
+    a = a.shear(g1=0.2, g2=-0.2).shift(1.1, -0.2).dilate(0.7)
+    b = b.shear(g1=0.2, g2=-0.2).shift(1.1, -0.2).dilate(0.7)
+    img_a = a.drawImage()
+    img_b = b.drawImage(img_a.copy())
+    # Fails at 6th decimal
+    np.testing.assert_array_almost_equal(img_a.array, img_b.array, 5,
+                                         "Transformed InterpolatedKImage image drawn incorrectly.")
```

Comment:
Maybe worth adding a test of an InterpolatedKImage convolved by something else?  Just to be more complete in the ways people will use the class.